pymc_marketing/mlflow.py

```diff
@@ -408,14 +424,362 @@ def log_inference_data(
+    include_last_observations : bool, optional
+        Whether to include the last observations of training data for adstock transformation.
+        Assumes X_pred are next predictions following training data. Used for all prediction
+        methods. Defaults to False.
+    original_scale : bool, optional
+        Whether to return predictions in original scale of target variable. Used for all
+        prediction methods. Defaults to True.
+
+    Notes
+    -----
+    This function logs the model as a native MLflow model, this is different to the full model object,
+    which includes the InferenceData. Doing this allows for the model to be stored in the MLFlow registry,
+    helping with model versioning and deployment.
+
+    Examples
+    --------
+    MLFlow Registering for a PyMC-Marketing MMM:
+
+    .. code-block:: python
+
+        import pandas as pd
+
+        import mlflow
+
+        from pymc_marketing.mmm import (
+            GeometricAdstock,
+            LogisticSaturation,
+            MMM,
+        )
+        import pymc_marketing.mlflow
+        from pymc_marketing.mlflow import log_mmm
+
+        pymc_marketing.mlflow.autolog(log_mmm=True)
+
+        # Usual PyMC-Marketing model code
+
+        data_url = "https://raw.githubusercontent.com/pymc-labs/pymc-marketing/main/data/mmm_example.csv"
+        data = pd.read_csv(data_url, parse_dates=["date_week"])
+
+        X = data.drop("y",axis=1)
+        y = data["y"]
+
+        mmm = MMM(
+            adstock=GeometricAdstock(l_max=8),
+            saturation=LogisticSaturation(),
+            date_column="date_week",
+            channel_columns=["x1", "x2"],
+            control_columns=[
+                "event_1",
+                "event_2",
+                "t",
+            ],
+            yearly_seasonality=2,
+        )
+
+        mlflow.set_experiment("MMM Experiment")
+
+        with mlflow.start_run():
+            idata = mmm.fit(X, y)
+
+            # Additional specific logging
+            fig = mmm.plot_components_contributions()
+            mlflow.log_figure(fig, "components.png")
+
+            model_info = log_mmm(
+                mmm=mmm,
+                registered_model_name="my_amazing_mmm",
+                include_last_observations=True,
+                original_scale=False,
+            )
+    """
+    # Incorporate MMM into MLflow workflow
+    mlflow_mmm = MMMWrapper(
+        model=mmm,
+        extend_idata=extend_idata,
+        combined=combined,
+        include_last_observations=include_last_observations,
+        original_scale=original_scale,
+    )
+
+    mlflow.pyfunc.log_model(
+        artifact_path=artifact_path,
+        python_model=mlflow_mmm,
+    )
+    run_id = mlflow.active_run().info.run_id
+    model_uri = f"runs:/{run_id}/{artifact_path}"
+
+    if registered_model_name:
+        mlflow.register_model(model_uri, registered_model_name)
+
+
+def load_mmm(
+    run_id: str,
```

Comment:
I'd rather this take `model_uri` which is more similar to the sklearn mlflow function: https://mlflow.org/docs/latest/python_api/mlflow.sklearn.html#mlflow.sklearn.load_model

This would allow loading from run or from registered model.

If we'd like, we can add helpers to construct the run URI and the registered model URI